extensions/amp-base-carousel/amp-base-carousel.md

```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
+  <button id='prev-button'>Go to previous slide</button>
+  <button id='next-button'>Go to next slide</button>
+  <button id='go-to-button'>Go to slide with green gradient</button>
+</div>
+
+<script>
+  (async () => {
+    const carousel = document.querySelector('#my-carousel');
+    await customElements.whenDefined('amp-base-carousel');
+    const api = await carousel.getApi();
+
+    // programatically advance to next slide
+    api.next();
+
+    // set up button actions
+    document.querySelector('#prev-button').onclick = () => api.prev();
+    document.querySelector('#next-button').onclick = () => api.next();
+    document.querySelector('#go-to-button').onclick = () => api.goToSlide(2);
+  })();
+</script>
+```
+
+#### Interactivity and API usage
+
+Bento enabled components in standalone use are highly interactive through their API. In Bento standalone use, the element's API replaces AMP Actions and events and [`amp-bind`](https://amp.dev/documentation/components/amp-bind/?format=websites).
+
+The `amp-base-carousel` component API is accessible by including the following script tag in your document:
+
+```
+await customElements.whenDefined('amp-base-carousel');
+const api = await carousel.getApi();
+```
+
+##### Actions
+
+The `amp-base-carousel` API allows you to perform the following actions:
+
+**next()**
+Moves the carousel forwards by `advance-count` slides.
+
+```
+api.next();
+```
+
+**prev()**
+Moves the carousel backwards by `advance-count` slides.
+
+```
+api.prev();
+```
+
+**goToSlide(index: number)**
+Moves the carousel to the slide specified by the `index` argument.
+
+```
+api.goToSlide(0); // Advance to first slide.
+api.goToSlide(length - 1); // Advance to last slide.
+```
+
+##### Events
+
+The `amp-base-carousel` API allows you to register and respond to the following events:
+
+**slideChange**
+
+This event is triggered when the index displayed by the carousel has changed.
+The new index is available via `event.data.index`.
+
+```
+carousel.addEventListener('slideChange', (e) => console.log(e.data.index))
+```
+
+#### Layout and style
+
+Each Bento component has a small CSS library you must include to guarantee proper loading without [content shifts](https://web.dev/cls/). Because of order-based specificity, you must manually ensure that stylesheets are included before any custom styles.
+
+```
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.css">
+```
+
+Fully valid AMP pages use the AMP layout system to infer sizing of elements to create a page structure before downloading any remote resources. However, Bento use imports components into less controlled environments and AMP's layout system is inaccessible.
+
+**Container type**
+
+The `amp-base-carousel` component has a defined layout size type. To ensure the component renders correctly, be sure to apply a size to the component and its immediate children (slides) via `height` and `width` or `aspect-ratio`:
```

Comment:
Maybe "via a desired CSS layout (such as `height`, `width`, `aspect-ratio`, and other)"? Just want to stress that any CSS layout rules would work.

Reply:
Done.